compiler/executable/insert/executable.rs

```diff
@@ -57,9 +58,15 @@ pub fn compile(
     input_variables: &HashMap<Variable, VariablePosition>,
     type_annotations: &TypeAnnotations,
     variable_registry: &VariableRegistry,
+    desired_output_variable_positions: Option<HashMap<Variable, VariablePosition>>,
     source_span: Option<Span>,
 ) -> Result<InsertExecutable, Box<WriteCompilationError>> {
-    let mut variable_positions = input_variables.clone();
+    debug_assert!(desired_output_variable_positions
+        .clone()
+        .map(|positions| { input_variables.iter().all(|(k, v)| positions.get(k).unwrap() == v) })
+        .unwrap_or(true));
```

Comment:
This assumption is made by pure `insert` stages. Though we pass `None` to them now.
Violating this assumption in a `put` stage should be fine.